wideq/client.py

```diff
@@ -304,53 +306,62 @@ class ModelInfo(object):
     def __init__(self, data):
         self.data = data
 
-    def value(self, name):
+    def value(self, name: str):
         """Look up information about a value.
 
-        Return either an `EnumValue` or a `RangeValue`.
+        :param name: The name to look up.
+        :returns: One of (`BitValue`, `EnumValue`, `RangeValue`,
+            `ReferenceValue`).
+        :raises ValueError: If an unsupported type is encountered.
         """
         d = self.data['Value'][name]
         if d['type'] in ('Enum', 'enum'):
             return EnumValue(d['option'])
         elif d['type'] == 'Range':
             return RangeValue(
-                d['option']['min'], d['option']['max'], d['option']['step']
+                d['option']['min'], d['option']['max'],
+                d['option'].get('step', 1)
             )
+        elif d['type'].lower() == 'bit':
+            bit_values = {
+                opt['startbit']: {
+                    'length': opt['length'],
+                    'value': opt['value'],
+                } for opt in d['option']
+            }
+            return BitValue(bit_values)
+        elif d['type'].lower() == 'reference':
+            return ReferenceValue(d['option'][0])
         else:
-            assert False, "unsupported value type {}".format(d['type'])
+            raise ValueError("unsupported value type {}".format(d['type']))
```

Comment:
Maybe a `NotImplementedError` would be slightly more appropriate?

Reply:
I think `ValueError` is more appropriate here as it's an unexpected value.  `NotImplementedError` to me is more aimed at developers using a library and not implementing a particular method in a class that is abstract or meant to be more like an interface.